### PR TITLE
client: remove wrapResponseError()

### PR DESCRIPTION
**Pull request by thaJeztah**
The wrapResponseError() utility converted some specific errors, but in doing so, could hide the actual error message returned by the daemon. In addition, starting with 38e6d474affe08230043fa36fa02f623c2ab2661 (https://github.com/moby/moby/pull/38689), HTTP status codes were already mapped to their corresponding errdefs types on the client-side, making this conversion redundant.

This patch removes the wrapResponseError() utility; it's worth noting that some error-messages will change slightly (as they now return the error as returned by the daemon), but may cointain more details as before, and in some cases prevents hiding the actual error.

Before this change:

```bash
docker container rm nosuchcontainer
Error: No such container: nosuchcontainer

docker container cp mycontainer:/no/such/path .
Error: No such container:path: mycontainer:/no/such/path

docker container cp ./Dockerfile mycontainer:/no/such/path
Error: No such container:path: mycontainer:/no/such

docker image rm nosuchimage
Error: No such image: nosuchimage

docker network rm nosuchnetwork
Error: No such network: nosuchnetwork

docker volume rm nosuchvolume
Error: No such volume: nosuchvolume

docker plugin rm nosuchplugin
Error: No such plugin: nosuchplugin

docker checkpoint rm nosuchcontainer nosuchcheckpoint
Error response from daemon: No such container: nosuchcontainer

docker checkpoint rm mycontainer nosuchcheckpoint
Error response from daemon: checkpoint nosuchcheckpoint does not exist for container mycontainer    

docker service rm nosuchservice
Error: No such service: nosuchservice

docker node rm nosuchnode
Error: No such node: nosuchnode

docker config rm nosuschconfig
Error: No such config: nosuschconfig

docker secret rm nosuchsecret
Error: No such secret: nosuchsecret
```

After this change:

```bash
docker container rm nosuchcontainer
Error response from daemon: No such container: nosuchcontainer

docker container cp mycontainer:/no/such/path .
Error response from daemon: Could not find the file /no/such/path in container mycontainer

docker container cp ./Dockerfile mycontainer:/no/such/path
Error response from daemon: Could not find the file /no/such in container mycontainer

docker image rm nosuchimage
Error response from daemon: No such image: nosuchimage:latest

docker network rm nosuchnetwork
Error response from daemon: network nosuchnetwork not found

docker volume rm nosuchvolume
Error response from daemon: get nosuchvolume: no such volume

docker plugin rm nosuchplugin
Error response from daemon: plugin "nosuchplugin" not found

docker checkpoint rm nosuchcontainer nosuchcheckpoint
Error response from daemon: No such container: nosuchcontainer

docker checkpoint rm mycontainer nosuchcheckpoint
Error response from daemon: checkpoint nosuchcheckpoint does not exist for container mycontainer

docker service rm nosuchservice
Error response from daemon: service nosuchservice not found

docker node rm nosuchnode
Error response from daemon: node nosuchnode not found

docker config rm nosuchconfig
Error response from daemon: config nosuchconfig not found

docker secret rm nosuchsecret
Error response from daemon: secret nosuchsecret not found
```

**- Description for the changelog**

```
Prevent the client from hiding error-details as returned by the server.
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
OK, couple of tests to update, as they were doing some string matching;

```
=== RUN   TestCopyFromContainerPathDoesNotExist
    copy_test.go:32: assertion failed: expected error to contain "No such container:path: 515efc53340e99efe9729e63e42ca484c4c3a8dfd797eb6840a1e389a2d18e01:/dne", got "Error response from daemon: Could not find the file /dne in container 515efc53340e99efe9729e63e42ca484c4c3a8dfd797eb6840a1e389a2d18e01"
        Error response from daemon: Could not find the file /dne in container 515efc53340e99efe9729e63e42ca484c4c3a8dfd797eb6840a1e389a2d18e01
--- FAIL: TestCopyFromContainerPathDoesNotExist (0.10s)

=== RUN   TestCopyToContainerPathDoesNotExist
    copy_test.go:62: assertion failed: expected error to contain "No such container:path: e7240951ade4a20ab965ae9dd8cb96dcc4d8b344fa7da39c313477561c3d014e:/dne", got "Error response from daemon: Could not find the file /dne in container e7240951ade4a20ab965ae9dd8cb96dcc4d8b344fa7da39c313477561c3d014e"
        Error response from daemon: Could not find the file /dne in container e7240951ade4a20ab965ae9dd8cb96dcc4d8b344fa7da39c313477561c3d014e
--- FAIL: TestCopyToContainerPathDoesNotExist (0.06s)
```

This one is something we should look at; there's also some code in docker/cli to strip prefixes etc; having a test to verify we don't panic is good, but we should also handle empty names in the client and mark them as invalid;

```
=== RUN   TestDockerSuite/TestContainerAPIDeleteWithEmptyName
    docker_api_containers_test.go:1593: assertion failed: expected error to contain "No such container", got "Error response from daemon: page not found"
        Error response from daemon: page not found
    --- FAIL: TestDockerSuite/TestContainerAPIDeleteWithEmptyName (0.00s)
```

**Comment by thaJeztah** [comment]
@ndeloof @rumpl @tianon ptal 🤗

**Review by tao12345666333 — APPROVED** [review]
LGTM

File: client/errors.go [hunk 1]
```diff
@@ -59,19 +58,6 @@ func (e objectNotFoundError) Error() string {
...
 }
 
-func wrapResponseError(err error, resp serverResponse, object, id string) error {
-	switch {
-	case err == nil:
-		return nil
-	case resp.statusCode == http.StatusNotFound:
-		return objectNotFoundError{object: object, id: id}
-	case resp.statusCode == http.StatusNotImplemented:
-		return errdefs.NotImplemented(err)
```
Comment (tianon) [inline client/errors.go:69]: I 100% agree the 404 handling here is less than ideal (and can see how it would hide the error, since it's not part of the new object), but I worry about this `errdefs.NotImplemented` handling -- are there other users of this client that might be affected by this negatively and no longer accurately detect when the answer they get back is a "Not Implemented" answer?

**Comment by thaJeztah** [comment]
> Prevent the client from bidding error-details as returned by the server.

I'll fix the typo in this description later (editing will re-trigger CI 😂)

**Review by tianon — APPROVED** [review]
Generally seems OK to me, just one minor concern (that's maybe not even relevant/important?)

File: client/errors.go (hunk 1 above)
Comment (tianon) [reply client/errors.go:69]: Ah, I'm satisfied! :heart:

**Review by cpuguy83 — APPROVED** [review]
LGTM
